soup_test.py

```diff
@@ -0,0 +1,33 @@
+from bs4 import BeautifulSoup
+from urllib.request import urlopen
+
+url = "https://lpi.oregonstate.edu/mic/glossary" #specify source location
+page = urlopen(url)
+html = page.read().decode("utf-8")
+soup = BeautifulSoup(html, "html.parser")
+
+
+glossary = {}
+sections = soup.find_all('dl')
+
+for section in sections:
+    words = section.find_all('dt')
+    for word in words:
+        check = word.find_next('dd')
+        definition = check.text.strip()
+        while True:
```

Comment:
I'm assuming it has since you've been able to make this work, but has this been tested and the loop concludes as is? If so then there isn't much to be concerned about. Having a potential infinite loop in something like this could be bad, though there aren't many variables that could cause this to become one if it current works as is.